modules/heat_conduction/src/dirackernels/GapHeatPointSourceMaster.C

```diff
@@ -47,12 +50,27 @@ GapHeatPointSourceMaster::GapHeatPointSourceMaster(const InputParameters & param
   if (parameters.isParamValid("tangential_tolerance"))
     _penetration_locator.setTangentialTolerance(getParam<Real>("tangential_tolerance"));
 
-  if (parameters.isParamValid("normal_smoothing_distance"))
-    _penetration_locator.setNormalSmoothingDistance(getParam<Real>("normal_smoothing_distance"));
-
-  if (parameters.isParamValid("normal_smoothing_method"))
-    _penetration_locator.setNormalSmoothingMethod(
-        parameters.get<std::string>("normal_smoothing_method"));
+  if (isParamValid("normal_smoothing_method"))
+  {
+    MooseEnum smoothing_method = getParam<MooseEnum>("normal_smoothing_method");
+    if (smoothing_method == "edge_based")
+    {
+      if (!isParamValid("normal_smoothing_distance"))
+        mooseError(
+            name(),
+            ": For edge based normal smoothing, normal_smoothing_distance parameter must be set.");
+      _penetration_locator.setEdgeBaseSmoothingMethod(getParam<Real>("normal_smoothing_distance"));
+    }
+    else if (smoothing_method == "nodal_normal_based")
+    {
+      if (!isParamValid("nodal_normals"))
+        mooseError(name(),
+                   ": For nodal nodal based smoothing, nodal_normals parameter must be set.");
+      _penetration_locator.setNodalNormalSmoothingMethod(getParam<UserObjectName>("nodal_normals"));
+    }
+    else
+      mooseError(name(), ": Unknown smoothing method.");
```

Comment:
This same error checking code is in at least 3 places in this PR. Maybe factor out into a single method?

Reply:
Yes, I hate this too. Even the original code had the same flaw. I will refactor that in a separate commit